job_engine/connector/kubernetes.py

```diff
@@ -733,6 +733,48 @@ def reboot(self):
             log.debug(f'The result of the reboot action: {reboot_result}')
         return reboot_result
 
+    @property
+    def deployed_components(self):
+        """
+        Function to get the installed components
+        """
+        log.debug("Calling for components...\n")
+        resources = \
+            self.api.search('nuvlabox-status', \
+                filter=f"parent=\"{self.nuvlabox_id}\" ", \
+                ).resources
+        for resource in resources:
+            nuvlabox_status_string = str(resource.id)
+            log.debug(f"Nuvlabox status string: {nuvlabox_status_string}")
+            nb_status_data = self.api.get(nuvlabox_status_string).data
+            log.debug(f"The nuvlabox-status data is:\n{json.dumps(nb_status_data, indent=2)}")
```

Comment:
I would suggest not using f-string and provide json serialised string as param. Also, the json serialisation is heavy, so hiding everything behind the log level check would be better.              
```python
if log.level == logging.DEBUG:
   log.debug("The nuvlabox-status data is:\n %s",
                       json.dumps(nb_status_data, indent=2))
```

Reply:
I have changed this at this point and also above where JSON is dumped.